### PR TITLE
Refactor @faststore/cli directory.ts to enable it to receive a basepath arg

### PR DESCRIPTION
**Pull request by gvc**
## What's the purpose of this pull request?

The first 4 commits were to better delineate the surface of the `directory.ts` util, I've exported only things being used outside of it. I've also renamed some variables (some variables referred to files and had `Dir` as a suffix).

## How to test it?

This PR adds NO behavioral change.

### CODE REVIEW

File: packages/cli/src/utils/directory.test.ts
Comment (eduardoformiga) [inline packages/cli/src/utils/directory.test.ts:1]: 💯

**Review by eduardoformiga — APPROVED** [review]

File: packages/cli/src/utils/directory.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  })
+
+  describe('tmpDir', () => {
+    it('is the cwd + .faststore', () => {
+      expect(pathsToMatch(tmpDir, './.faststore')).toBe(true)
+    })
+  })
+
+  describe('userDir', () => {
+    it('returns the directory of the starters package.json', () => {
```
Comment (lariciamota) [inline packages/cli/src/utils/directory.test.ts:25]:
```suggestion
    it("returns the directory of the starter's package.json", () => {
```

File: packages/cli/src/utils/directory.test.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  })
+
+  describe('tmpCustomizationsSrcDir', () => {
+    it('returns the directory of customizations directory on the tmp dir', () => {
+      expect(pathsToMatch(tmpCustomizationsSrcDir, './.faststore/src/customizations/src')).toBe(true)
+    })
+  })
+
+  describe('userThemesFileDir', () => {
+    it('returns the directory of the starters package.json', () => {
```
Comment (lariciamota) [inline packages/cli/src/utils/directory.test.ts:37]:
```suggestion
    it("returns the directory of the starter's themes", () => {
```

File: packages/cli/src/utils/directory.test.ts
Comment (lariciamota) [reply packages/cli/src/utils/directory.test.ts:1]: Nice!! It's great to have tests for this :)

File: packages/cli/src/utils/directory.test.ts (hunk 1 above)
Comment (gvc) [reply packages/cli/src/utils/directory.test.ts:25]: I was too lazy to do that. Thank you :pray:

**Review by lariciamota — APPROVED** [review]

File: packages/cli/src/utils/directory.test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+const pathsToMatch = (expected: string, desired: string) => {
+  const expectedResolved = path.resolve(expected)
+  const desiredResolved = path.resolve(desired)
+
+  return expectedResolved === desiredResolved
+}
+
+describe('directory with base path', () => {
+  describe('coreDir', () => {
+    it('is the fastsstoreDir + core', () => {
```
Comment (vaporwavie) [inline packages/cli/src/utils/directory.test.ts:13]: @gvc typo: `fastsstoreDir` should be `faststoreDir`

Comment (gvc) [reply packages/cli/src/utils/directory.test.ts]: the eyes of an eagle! Thank you!

**Pull request merged**